### ISSUE TITLE
Ignore case - case insensitive "rename" - any_Case-IGNOREcase

### ISSUE THREAD

**Issue by firstdorsal** [serde-rs/serde#2359]
# Idea
Similar to `#[serde(rename_all = "camelCase")]` I would love to see a `#[serde(rename_all = "any_Case-IGNOREcase")]` that ignores casing accepting any case as long as the characters match.

# Use case
I am parsing labels for the reverse proxy traefik and they can have at least camel case and lowercase as far as I can see making it a bit difficult to parse them into a structure with serde.

# Examples

```rs
#[derive(Deserialize, Debug, Clone)]
#[serde(rename_all = "any_Case-IGNOREcase")]
pub struct DockerLabelsProvider {
    pub exposed_by_default: Option<bool>,
}
```
Valid:
`exposedByDefault`
`EXPOSEDBYDEFAULT`
`exposed_by_default`
`exposed-by-default`
`exposedbydefault`

Invalid:
`exposed_BEI_default`

Thank you for creating serde. It's awesome! :)

**Comment by j-a-m-l** [comment]
Maybe you could use https://docs.rs/serde-aux/latest/serde_aux/index.html.

My current approach is:

```
#[derive(Deserialize, Debug, Clone)]
pub struct DockerLabelsProvider {
    #[serde(alias = "exposedByDefault", alias = "EXPOSED_BY_DEFAULT")]
    pub exposed_by_default: Option<bool>,
}
```
This approach admits all aliases and the name of the struct field, but is not case insensitive.

**Comment by firstdorsal** [comment]
Thank you for the suggestion, but this is not an option as I would have to do this for every option and field in a large config file.

**Comment by j-a-m-l** [comment]
Have you tried https://docs.rs/serde-aux/latest/serde_aux/container_attributes/fn.deserialize_struct_case_insensitive.html?

Another simple solution is something like https://github.com/serde-rs/serde/issues/586 and then convert the `Map` to your struct.

Alternatively you'd need to implement `Deserializer`: https://serde.rs/impl-deserializer.html

**Comment by dtolnay** [comment]
I think I would prefer not to build this into serde_derive. But it would be reasonable for someone to maintain a more fully featured derive macro that supports this, or a library that solves this via some kind of Deserializer adapter, as suggested above.

**Comment by Mingun** [comment]
Actually, this problem could be fixed `alias_all` container attribute from #1530
